src/exo/internal_cursors.py

```diff
@@ -271,6 +280,24 @@ def __len__(self):
         _, _range = self._path[-1]
         return len(_range)
 
+    # ------------------------------------------------------------------------ #
+    # Block-specific operations
+    # ------------------------------------------------------------------------ #
+
+    def expand(self, lo=None, hi=None):
```

Comment:
I think it'll be useful if we can do a syntax sugar on this s.t.
```
c = cblock.next()
cblock = cblock + c
```

Reply:
Yes, there may be better choices.  (I would use the union operator `a | b` for what you're suggesting, not `+`)  This "expand" method is just there so I could solve one particular issue and move forward.  Don't take it as a long term commitment for what the API should be.

Reply:
Make sense. Shouldn't it be `+` because it's like expanding a list? union feels like unordered.

Reply:
It's a join in an appropriate lattice of sequences really.  For example, suppose I have cursors to two statements with another statement in between.
```
s1 <- c1
s2
s3 <- c2
```

The smallest block containing the two statements with cursors (`s1` and `s3`) is a block of all 3 statements.  That's neither a concatenation nor union.  It's a join of "intervals."